checker/cost_test.go

```diff
@@ -503,6 +503,31 @@ func TestCost(t *testing.T) {
 			},
 			wanted: CostEstimate{Min: 3, Max: 3},
 		},
+		{
+			name:   ".filter list literal",
+			expr:   `[1,2,3,4,5].filter(x, x % 2 == 0)`,
+			wanted: CostEstimate{Min: 41, Max: 101},
+		},
+		{
+			name:   ".map list literal",
+			expr:   `[1,2,3,4,5].map(x, x)`,
+			wanted: CostEstimate{Min: 86, Max: 86},
+		},
+		{
+			name:   ".map.filter list literal",
+			expr:   `[1,2,3,4,5].map(x, x).filter(x, x % 2 == 0)`,
+			wanted: CostEstimate{Min: 117, Max: 177},
+		},
+		{
+			name:   ".map.exists list literal",
+			expr:   `[1,2,3,4,5].map(x, x).exists(x, x % 2 == 0) == true`,
+			wanted: CostEstimate{Min: 108, Max: 123},
+		},
+		{
+			name:   ".map.map list literal",
+			expr:   `[1,2,3,4,5].map(x, x).map(x, x)`,
```

Comment:
Do we need to check the return type of the expression in `map()`? What about something like: 

```
[1,2,3,4,5].map(x, [x, x])
```

Reply:
We have a gap today where we don't track composed (nested) sized objects well.  I think we need to do better, but it would be a larger than than this fix.

Reply:
The runtime calculations would be a backstop if we end up undershooting the cost here, right?

Reply:
Runtime cost tallies actual operations as they execute, so it's a really good way to make sure we don't overuse CPU even if expected cost checking lets an expression in that's too expensive.

Reply:
Good idea. Added.

Reply:
@jpbetz even if we don't have confidence in the correctness of `[1,2,3,4,5].map(x, [x, x])`, we should add a test case to document the current behavior and make sure we understand the impact of the fix.

Reply:
Added. As expected, there is a variation of this that fails to compute a bounded estimated cost. I've included a test showing it along with an explanation.